include/deal.II/grid/tria.h

```diff
@@ -511,6 +513,167 @@ namespace internal
 /*------------------------------------------------------------------------*/
 
 
+namespace construction
```

Comment:
same comment as above for the namespace name, of course.

I'd just break this part out into a file of its own, for simplicity, and merge it with the separate file you have for the utilities

Reply:
The issue here that `ConstructionData` depends on `CellData` and `Triangulation` on `ConstructionData` and the `ConstructionData`-utility functions on `Triangulation`...

Reply:
Couldn't you get away with forward declarations in most cases?

Reply:
I have moved `TriangulationDescription::Description` and the other data structures to the new file. However, I have also moved `CellData` and `SubCellData` to that file. The file `tria.h` includes now `tria_description.h`; and `tria_description.h` includes `tria.h`; this cyclic dependency is a real mess.

> Couldn't you get away with forward declarations in most cases?

As a side remark,  I consider forward declarations as a pure evil "feature".